swift_scality_backend/diskfile.py

```diff
@@ -178,7 +178,8 @@ def unexpected_http_status(response):
             conn = swift.common.bufferedhttp.http_connect_raw(
                 address, port, method, safe_path, headers)
 
-        with contextlib.closing(conn), eventlet.Timeout(self.proxy_timeout):
+        with contextlib.nested(contextlib.closing(conn),
```

Comment:
I am not sure about the order of these 2 context manager. And I may have missed the warning about the nested() method in python documentation : https://docs.python.org/2/library/contextlib.html#contextlib.nested